File: GGAndroid_2.0/GGCore/GGLib/src/main/java/io/ganguo/library/ui/fragment/BaseFragment.java
package io.ganguo.library.ui.fragment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import io.ganguo.library.R;
import io.ganguo.utils.common.UIHelper;
import io.ganguo.utils.util.log.Logger;

/**
 * Fragment - 基类
 * <p/>
 * Created by zhihui_chen on 14-8-4.
 */
public abstract class BaseFragment extends Fragment implements InitResources {

    public String getFragmentTitle() {
        return null;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        Logger.d("onCreateView");
        return inflater.inflate(getLayoutResourceId(), container, false);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        // bind back action
        UIHelper.bindActionBack(getActivity(), getView().findViewById(R.id.action_back));

        // init resources
        initView();
        initListener();
        initData();

        Logger.d("onActivityCreated");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        Logger.i("onDestroy");
    }
}

File: GGAndroid_3.0/core/support/core/src/main/java/io/ganguo/core/ui/fragment/BaseFragment.kt
package io.ganguo.core.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment


/**
 * <pre>
 *     author : leo
 *     time   : 2019/10/11
 *     desc   : Fragment - 基类
 * </pre>
 */
abstract class BaseFragment : Fragment(), InitResources {
    var fragmentTitle: String = ""

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(layoutResourceId, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        // init resources
        initView()
        initListener()
        initData()
    }

}